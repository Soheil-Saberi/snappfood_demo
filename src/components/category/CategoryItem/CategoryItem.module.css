.a{
    @apply flex flex-col items-center p-3 border rounded-md shadow-md space-y-2
}

.a:hover{
    @apply bg-gray-100
}

.a img{
    @apply w-3/4
}

.a p {
    @apply text-center text-xs
}