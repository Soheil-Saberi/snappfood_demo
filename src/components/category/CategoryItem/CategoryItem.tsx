import { Fragment } from "react";
// Style
import styles from "./CategoryItem.module.css";

export default function CategoryItem(props: any) {
  const { id, title, img, alt } = props;
  return (
    <Fragment key={id}>
      <a href={`http://localhost:3000/#${id}`} className={styles.a}>
        <img src={img} alt={alt} />
        <p>{title}</p>
      </a>
    </Fragment>
  );
}
