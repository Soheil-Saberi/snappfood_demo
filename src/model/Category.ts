export default interface Category {
  id: number;
  title: string;
  img: string;
  alt: string;
}
