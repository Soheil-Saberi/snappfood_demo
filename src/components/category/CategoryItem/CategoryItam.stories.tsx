import { ComponentMeta, ComponentStory } from '@storybook/react';
import Category from '../../../model/Category';
import CategoryValues from '../../../utility/CategoryValues';
import CategoryItem from './CategoryItem';

interface Args {
  category: Category;
}

export default {
  component: CategoryItem,
  title: 'Category/Item'
} as ComponentMeta<typeof CategoryItem>;

export const CategoryItemStory: ComponentStory<typeof CategoryItem> = (args: Args) => (
  <CategoryItem category={args.category} />
);

CategoryItemStory.args = {
  category: CategoryValues[0]
};
