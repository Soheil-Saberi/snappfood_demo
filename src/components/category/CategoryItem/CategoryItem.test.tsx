import renderer from 'react-test-renderer';
import { describe, expect, test } from '@jest/globals';
import { cleanup } from '@testing-library/react';
import CategoryItem from './CategoryItem';
import Category from '../../../model/Category';

afterEach(() => {
  cleanup();
});

describe('CategoryItem Component', () => {
  test('Component Rendernig With Value Template', () => {
    const categoryValue: Category = {
      id: 1,
      title: 'پیتزا',
      img: '../assets/images/pizza.png',
      alt: 'pizza-icon'
    };
    const categoryItem = renderer.create(<CategoryItem category={categoryValue} />);
    expect(categoryItem.toJSON()).toMatchSnapshot();
  });

  test('Component Rendernig With Empty title', () => {
    const categoryValue: Category = {
      id: 1,
      title: '',
      img: '../assets/images/pizza.png',
      alt: 'pizza-icon'
    };
    const categoryItem = renderer.create(<CategoryItem category={categoryValue} />);
    expect(categoryItem.toJSON()).toBeNull();
  });
});
