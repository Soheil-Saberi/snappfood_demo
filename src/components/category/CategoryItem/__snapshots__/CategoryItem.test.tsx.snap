// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem Component Component Rendernig With Value Template 1`] = `
<a
  href="undefined/#1"
>
  <img
    alt="pizza-icon"
    src="../assets/images/pizza.png"
  />
  <p>
    پیتزا
  </p>
</a>
`;
